## Analytical tools and pipelines

Pipelines for processing and analyzing data from epigenomic assays and integrating with genetic association data  

&nbsp;

## Epigenome database
Instructions for deploying and depositing data into epigenome database DGA based on ENCODE:  
```
DGA/ 
```

&nbsp;  
  
## ATAC-seq
Process bulk (i.e. non-single cell) ATAC-seq data: 
```
bulk_ATAC-seq/bulk_ATAC-seq_pipeline.py 
```

&nbsp;  

ENCODE ATAC-seq pipeline:  
https://github.com/ENCODE-DCC/atac-seq-pipeline  

&nbsp;

## ChIP-seq
Process TF or histone modification ChIP-seq data:
```
ChIP-seq/ChIP-seq_pipeline.py  
```

&nbsp;  

ENCODE ChIP-seq pipeline:  
https://github.com/ENCODE-DCC/chip-seq-pipeline2  

&nbsp;  

## RNA-seq

ENCODE RNA-seq pipeline:  
https://github.com/ENCODE-DCC/rna-seq-pipeline  

&nbsp;  

## CUT&RUN
Process TF or histone modification CUT&RUN data:
```
CUTNRUN/cut_and_run_pipeline.py  
```

&nbsp;  

## PLAC-seq
Process PLAC-seq/HiChiP data.
```
PLAC-seq/MAPS_pipeline.py
```

&nbsp;

## Single nuclear ATAC-seq
Process single nuclear ATAC-seq (snATAC-seq data)  

Pipeline used for human lung snATAC-seq data in Wang, Chiou et al bioRxiv:  
```
snATAC_seq/lung_snATAC_pipeline.py 
```
  
&nbsp;  

SnapATAC:  
https://github.com/r3fang/SnapATAC  

&nbsp;


